pkg/placement/controllers/scheduling/schedule.go

```diff
@@ -79,10 +84,26 @@ type PrioritizerResult struct {
 	Scores PrioritizerScore `json:"scores"`
 }
 
+// ScheduleDecisionGroups groups the cluster decisions by group strategy
+type ScheduleDecisionGroups []ScheduleDecisionGroup
+
+type ScheduleDecisionGroup struct {
+	DecisionGroupName string
+	ClusterDecisions  []clusterapiv1beta1.ClusterDecision
+}
+
+func (g *ScheduleDecisionGroups) Total() int {
+	num := 0
+	for _, group := range *g {
+		num += len(group.ClusterDecisions)
+	}
+	return num
+}
+
 // ScheduleResult is the result for a certain schedule.
 type scheduleResult struct {
 	feasibleClusters     []*clusterapiv1.ManagedCluster
-	scheduledDecisions   []clusterapiv1beta1.ClusterDecision
```

Comment:
I actually think the scheduledDecisions here should be []*clusterapiv1.ManagedCluster,

And you should have a process like 
1. schedule
2. generateDecision
3. bind

Reply:
Change to []*clusterapiv1.ManagedCluster and update the placement decision generate and update flow.